### DISCUSSION TITLE
Is AppFlowy replacing Flutter with Tauri?

### DISCUSSION THREAD

**Discussion by taviks** [AppFlowy-IO/AppFlowy#1746] [General]
I've recently been evaluating each and was intrigued to see recent Tauri-related commits and issues here. However, I couldn't find any discussion on whether the plan is to have Tauri completely replace Flutter or compliment it. 

Maybe the thinking is to use Flutter for mobile and Tauri for desktop? If so, why not Flutter for desktop?

**Comment by annieappflowy** [comment]
@appflowy answered on Discord. Copied his reply as below and let us know if you have follow-up questions.

I suppose we can support both frameworks. When we just started building AppFlowy, Tauri supporting desktop was a bit behind Flutter's progress. On top of it, we believe Flutter will remain a no-brainer for mobile development. Therefore, we chose Flutter over Tauri. 

Well, it's been over a year, and we find out Tauri has come out strong in desktop development lately, supporting more desktop features than Flutter https://github.com/tauri-apps/tauri#features

So we want to explore it and see if it can be a better choice for AppFlowy desktop version. The experiment wouldn't be costly, as we explained in our blog article, the data component would not have to change if the UI component switched from one platform to another.
https://blog-appflowy.ghost.io/tech-design-flutter-rust/

![image](https://user-images.githubusercontent.com/12026239/215254395-956c405a-61c2-4392-bda3-cf3dbea9eaa4.png)
